src/wasm-lib/kcl/src/parsing/parser.rs

```diff
@@ -742,21 +742,58 @@ pub(crate) fn array_elem_by_elem(i: &mut TokenSlice) -> PResult<Node<ArrayExpres
     )
     .context(expected("array contents, a list of elements (like [1, 2, 3])"))
     .parse_next(i)?;
+    ignore_trailing_comma(i);
     ignore_whitespace(i);
-    let end = close_bracket(i)
-        .map_err(|e| {
-            if let Some(mut err) = e.clone().into_inner() {
-                err.cause = Some(CompilationError::fatal(
-                    open.as_source_range(),
-                    "Array is missing a closing bracket(`]`)",
-                ));
-                ErrMode::Cut(err)
-            } else {
-                // ErrMode::Incomplete, not sure if it's actually possible to end up with this here
-                e
-            }
-        })?
-        .end;
+
+    let maybe_end = close_bracket(i).map_err(|e| {
+        if let Some(mut err) = e.clone().into_inner() {
+            let start_range = open.as_source_range();
+            let end_range = i.as_source_range();
+            err.cause = Some(CompilationError::fatal(
+                SourceRange::from([start_range.start(), end_range.start(), end_range.module_id().as_usize()]),
+                "Encountered an unexpected character(s) before finding a closing bracket(`]`) for the array",
+            ));
+            ErrMode::Cut(err)
+        } else {
+            // ErrMode::Incomplete, not sure if it's actually possible to end up with this here
+            e
+        }
+    });
+
+    if maybe_end.is_err() {
+        // if there is a closing bracket at some point, but it wasn't the next token, it's likely that they forgot a comma between some
+        // of the elements
+        let maybe_closing_bracket: PResult<((), Token)> = peek(repeat_till(
+            0..,
+            none_of(|token: Token| {
+                // bail out early if we encounter something that is for sure not allowed in an
+                // array, otherwise we could seek to find a closing bracket until the end of the
+                // file
+                RESERVED_WORDS
+                    .keys()
+                    .chain([",,", "{", "}", "["].iter())
+                    .any(|word| *word == token.value)
```

Comment:
This isn't super sophisticated, but it should be good enough to avoid accidentally parsing the whole file looking for closing brace/bracket